pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -530,8 +557,26 @@ func (r *Reconciler) ensureIsInstalled(ctx context.Context, log *zap.SugaredLogg
 		return fmt.Errorf("failed to create command: %w", err)
 	}
 
-	cmdLog := log.With("cmd", strings.Join(cmd.Args, " "))
+	if addon.Name == "csi" &&
+		cluster.Spec.Cloud.Hetzner != nil &&
+		cluster.Spec.Features[kubermaticv1.ClusterFeatureExternalCloudProvider] &&
+		cluster.Annotations[migratedHetznerCSIAddon] != "yes" {
```

Comment:
just a very minor edge-case nit, can you add a check for `cluster.Annotations` (not) being nil before we try to access it here so we don't run into this? It shouldn't happen, but I'm always uncomfortable accessing maps that can be nil.

```suggestion
		(cluster.Annotations == nil || cluster.Annotations[migratedHetznerCSIAddon] != "yes") {
```

Reply:
aren't these two statements semantically equivalent? Reading from nil map will return the default value for any key (so an empty string in this case).

Reply:
Sorry, you are right. Brain messed up, this only applies to write operations. Nvm!